internal/observerware/middleware_test.go

```diff
@@ -59,49 +61,113 @@ func TestUnaryInboundMiddleware(t *testing.T) {
 		wantFields []zapcore.Field
 	}{
 		{
-			desc:    "downstream errors",
+			desc:    "no downstream errors",
 			handler: fakeHandler{},
 			extract: NewNopContextExtractor(),
 			wantFields: []zapcore.Field{
-				zap.String("rpcType", "unary inbound"),
+				zap.String("rpcType", "unary"),
 				zap.Skip(),
-				zap.Object("request", req),
+				zap.Object("request", _req),
 				zap.Duration("latency", 0),
 				zap.Bool("successful", true),
 				zap.Skip(),
 			},
 		},
 		{
+			desc:    "downstream errors",
+			extract: NewNopContextExtractor(),
+			handler: fakeHandler{errFailed},
+			wantErr: true,
+			wantFields: []zapcore.Field{
+				zap.String("rpcType", "unary"),
+				zap.Skip(),
```

Comment:
I introduced skip fields just to make logging errors nicer - I'd like `zap.Error(err)` to do something sane even when `err == nil`, so that I don't have to use a separate conditional to choose whether to log the error message. Currently, it uses `zap.Skip()` to no-op nil errors.